src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function roundArea(area: number): number {
+  return (Math.trunc(area * 100) / 100);
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw Error('Triangle does not exist');
+    }
+
+    if (a < 0 || b < 0 || c < 0) {
+      throw Error('Input data is not valid');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
 
+    const p = (a + b + c) / 2;
+    const s = Math.sqrt(p * (p - a) * (p - b) * (p - c));
+
+    return roundArea(s);
+  }
 }
 
 export class Circle {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+    public shape: Shape = 'circle',
+  ) {
+    if (radius < 0) {
+      throw Error('Input data is not valid');
+    }
+  }
+
+  getArea(): number {
+    const s = Math.PI * (this.radius ** 2);
 
+    return roundArea(s);
+  }
 }
 
 export class Rectangle {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
```

Comment:
I think it could be better to use more descriptive names like `width` and `height`